src/data/fintype/card_embedding.lean

```diff
@@ -18,15 +18,13 @@ local notation `‖` x `‖` := fintype.card x
 
 open_locale nat
 
-local attribute [semireducible] function.embedding.fintype
-
 namespace fintype
 
--- We need the separate `fintype α` instance as it contains data,
--- and may not match definitionally with the instance coming from `unique.fintype`.
-lemma card_embedding_eq_of_unique
-  {α β : Type*} [unique α] [fintype α] [fintype β] [decidable_eq α] [decidable_eq β]:
-‖α ↪ β‖ = ‖β‖ := card_congr equiv.unique_embedding_equiv_result
+lemma card_embedding_eq_of_unique {α β : Type*} [unique α] [fintype β] [fintype (α ↪ β)] :
+  ‖α ↪ β‖ = ‖β‖ := card_congr equiv.unique_embedding_equiv_result
+
+-- needed for the below proof to work
+local attribute [semireducible] function.embedding.fintype
 
 private lemma card_embedding_aux {n : ℕ} {β} [fintype β] [decidable_eq β] (h : n ≤ ‖β‖) :
```

Comment:
can you put this `local attribute` in a section around just this proof? I hadn't realized the `irreducible` below is just restoring it back to the original reducibility.

Reply:
I'm going to push a much shorter proof.